### ISSUE TITLE
Catalina RQDA error

### ISSUE THREAD

**Issue by incedevrim** [JanMarvin/RQDA#9]
Thank you for macOs source package.  All loading was fine by launching RQDA.   R was aborted while launching. Some details, session and crash reports info etc., below. On the other hand I tried same package on Linux Ubuntu 20.04. It worked perfectly in first sight but saving projects give some errors that are about text style, UTF 16 etc., RSQLite or GTK-Critical**

```{r}
R version 4.0.1 (2020-06-06)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.5
Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
locale:
[1] C
attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     
loaded via a namespace (and not attached):
[1] compiler_4.0.1
```
   
This is error report:

```{r}
> library(RQDA)
Loading required package: RSQLite
Loading required package: gWidgets2RGtk2
Loading required package: memoise
Loading required package: RGtk2
Loading required package: gWidgets2
Loading required package: digest
Loading required package: DBI
Loading required package: stringi

Use 'RQDA()' to start the programme.


 *** caught segfault ***
address 0x0, cause 'memory not mapped'

Traceback:
 1: .RGtkCall("S_gtk_widget_set_visible", object, visible, PACKAGE = "RGtk2")
 2: method(obj, ...)
 3: widget$setVisible(as.logical(value))
 4: obj$set_visible(as.logical(value))
 5: `visible<-.default`(`*tmp*`, value = TRUE)
 6: `visible<-`(`*tmp*`, value = TRUE)
 7: RQDA()
 8: fun(libname, pkgname)
 9: doTryCatch(return(expr), name, parentenv, handler)
10: tryCatchOne(expr, names, parentenv, handlers[[1L]])
11: tryCatchList(expr, classes, parentenv, handlers)
12: tryCatch(fun(libname, pkgname), error = identity)
13: runHook(".onAttach", ns, dirname(nspath), nsname)
14: attachNamespace(ns, pos = pos, deps, exclude, include.only)
15: doTryCatch(return(expr), name, parentenv, handler)
16: tryCatchOne(expr, names, parentenv, handlers[[1L]])
17: tryCatchList(expr, classes, parentenv, handlers)
18: tryCatch({    attr(package, "LibPath") <- which.lib.loc    ns <- loadNamespace(package, lib.loc)    env <- attachNamespace(ns, pos = pos, deps, exclude, include.only)}, error = function(e) {    P <- if (!is.null(cc <- conditionCall(e)))         paste(" in", deparse(cc)[1L])    else ""    msg <- gettextf("package or namespace load failed for %s%s:\n %s",         sQuote(package), P, conditionMessage(e))    if (logical.return)         message(paste("Error:", msg), domain = NA)    else stop(msg, call. = FALSE, domain = NA)})
19: library(RQDA)

Possible actions:
1: abort (with core dump, if enabled)
2: normal R exit
3: exit R without saving workspace
4: exit R saving workspace
```

**Comment by JanMarvin** [comment]
Hi @incedevrim since I do not own any Apple device I cannot really help you with the Mac OS issue. Could you confirm that RGtk2 works on your system?

E.g. what happens if you run:  `library(RGtk2)`

Regarding your issues on Ubuntu:

> It worked perfectly in first sight but saving projects give some errors that are about text style, UTF 16 etc., RSQLite or GTK-Critical**

Since these are somewhat different errors could you open another issue and post error messages and provide examples to reproduce the errors?

**Comment by incedevrim** [comment]
Thank you for replies. library(RGtk2) seems working smoothly in R 4.0.1. I checked again in also RStudio 1.3959.
I will open another issue about Ubuntu 20.04.

**Comment by JanMarvin** [comment]
I did some research but apparently `RGtk2` is somehow broken on Apple (see https://github.com/lawremi/RGtk2/issues/2)

**Comment by incedevrim** [comment]
Yes it seems a chronic problem. Although the terminal speaks differently, I can understand that something is wrong with RQDA in its MacOS environment. I think I will install Ubuntu with a VM on MacBook or try dual-boot with Windows because I need urgently use RQDA. Using within Docker seems me so confusing regarding some settings of .env file etc. Also I am not sure whether the effort worths or not for a illiterate man on Docker. Thank you for interest.

**Comment by JanMarvin** [comment]
I think this can stay open as mac builds relying on CRAN binaries apparently are broken.

Did you by chance install as advised by the readme, installing xcode and building gtk2 (and Rgtk2 from source afterwards)? There is a slight chance that this will circumvent the issue, but most likely it'll take some time.

**Comment by incedevrim** [comment]
Yes I tried this version too and also via homebrew. As the last option I will remove completely R, RStudio, GTK2, Macports and other components. Then I will try one more later. Also I'll open two issues about RQDA running on MS Windows and on Ubuntu 20.04. The more stable one seems to me running on Linux, although there are a few errors that I will write about later.

**Comment by JanMarvin** [comment]
Kind of a bummer that it does not work on Apple, but from my understanding the issue is os related and has nothing to do with RQDA. I leave this issue open for others to pitch in and will remove the Mac part from the readme.
For now at least Linux and Windows virtual machines are available.

**Comment by incedevrim** [comment]
Thanks. It seems to me running on macOs Catalina at the now. I installed all packages from sources. I wrote about whole story in [another issue](https://github.com/Ronggui/RQDA/issues/38#issuecomment-651086544).

**Comment by JanMarvin** [comment]
Glad that you've got it working on Catalina. Though I'm not sure what your issues on Linux and Windows are about. There is no reason at all why root privileges are required or even helpful running RQDA on Linux or Windows (and both are somewhat tested).

**Comment by matthewjacraig** [comment]
Hi, I'm not sure if this has been covered yet or not but I am getting an error upon running
`> library(RQDA)`

This is what I get in return: 
`Loading required package: RSQLite`

`Error: package or namespace load failed for ‘RSQLite’ in loadNamespace(i, c(lib.loc, .libPaths()), versionCheck = vI[[i]]): namespace ‘rlang’ 0.4.6 is already loaded, but >= 0.4.7 is required`

`Error: package 'RSQLite' could not be loaded`

Any advice on resolving this would be great.
